crates/common/tedge_config/src/tedge_config_cli/tedge_config.rs

```diff
@@ -470,6 +472,17 @@ define_tedge_config! {
         #[doku(as = "PathBuf")]
         root_cert_path: Utf8PathBuf,
 
+        /// The authentication method used to connect Cumulocity
+        #[tedge_config(note = "In the auto mode, basic auth is used if c8y.credentials_path is set")]
+        #[tedge_config(example = "certificate", example = "basic", example = "auto", default(variable = AuthMethod::Certificate))]
```

Comment:
Keeping `certificate` as the default value makes the `auto` variant irrelevant IMO as I can't imagine it ever being used by anyone. When the default is already set to certificate, a user might either keep it or change it to basic. I can't imagine why someone would explicitly change that to `auto` and embrace uncertainty.